page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/DefaultPage.java

```diff
@@ -0,0 +1,168 @@
+import ru.sbtqa.tag.pagefactory.exceptions.PageException;
+import ru.sbtqa.tag.pagefactory.utils.ReflectionUtils;
+import ru.sbtqa.tag.qautils.errors.AutotestError;
+
+public abstract class DefaultPage extends Page {
+
+    PageActions pageActions = Environment.getPageActions();
+    PageChecks pageChecks = Environment.getPageChecks();
+
+    public DefaultPage(WebDriver driver) {
+        super(driver);
+    }
+
+    /**
+     * Find element with specified title annotation, and fill it with given text
+     * Add elementTitle-&gt;text as parameter-&gt;value to corresponding step in
+     * allure report
+     *
+     * @param elementTitle element to fill
+     * @param text text to enter
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.PageException if page was not
+     * initialized, or required element couldn't be found
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.fill.field")
+    public void fill(String elementTitle, String text) throws PageException {
+        Object element = ReflectionUtils.getElementByTitle(PageContext.getCurrentPage(), elementTitle);
+        pageActions.fill(element, text);
+    }
+
+    /**
+     * Find specified WebElement on a page, and click it Add corresponding
+     * parameter to allure report
+     *
+     * @param elementTitle title of the element to click
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.PageException if page was not
+     * initialized, or required element couldn't be found
+     */
+    @ActionTitles({
+            @ActionTitle("ru.sbtqa.tag.pagefactory.click.link"),
+            @ActionTitle("ru.sbtqa.tag.pagefactory.click.button")})
+    public void click(String elementTitle) throws PageException {
+        Object element = ReflectionUtils.getElementByTitle(PageContext.getCurrentPage(), elementTitle);
+        pageActions.click(element);
+    }
+
+    /**
+     * Press specified key on a keyboard Add corresponding parameter to allure
+     * report
+     *
+     * @param keyName name of the key. See available key names in {@link Keys}
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.press.key")
+    public void pressKey(String keyName) {
+        pageActions.press(null, keyName);
+    }
+
+    /**
+     * Focus a WebElement, and send specified key into it Add corresponding
+     * parameter to allure report
+     *
+     * @param keyName name of the key. See available key names in {@link Keys}
+     * @param elementTitle title of WebElement that accepts key commands
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.ElementNotFoundException if
+     * couldn't find element with required title
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.press.key")
+    public void pressKey(String keyName, String elementTitle) throws PageException {
+        Object element = ReflectionUtils.getElementByTitle(PageContext.getCurrentPage(), elementTitle);
+        pageActions.press(element, keyName);
+    }
+
+    /**
+     * TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)